pkg/controller/nodecsrapprover/node_csr_approver.go

```diff
@@ -59,19 +61,72 @@ type reconciler struct {
 }
 
 func Add(mgr manager.Manager) error {
-	certClient, err := certificatesv1client.NewForConfig(mgr.GetConfig())
+	// TODO: delete whole file node_csr_approver_v1beta1.go and dynamic API groups discovery
+	// after we drop kubernetes 1.18 support
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(mgr.GetConfig())
 	if err != nil {
-		return fmt.Errorf("failed to create certificate client: %v", err)
+		return fmt.Errorf("failed to init discovery client: %w", err)
 	}
 
-	r := &reconciler{Client: mgr.GetClient(), certClient: certClient.CertificateSigningRequests()}
-	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: r})
+	srvGroups, err := discoveryClient.ServerGroups()
+	if err != nil {
+		return fmt.Errorf("failed to server API groups: %w", err)
```

Comment:
Tiny nit: it feels like this error message is missing a word.

```suggestion
		return fmt.Errorf("failed to get server API groups: %w", err)
```

Reply:
indeed :D